lib/features/drawing_detail/drawing_detail_bloc.dart

```diff
@@ -129,4 +132,43 @@ class DrawingDetailBloc extends Bloc<DrawingDetailEvent, DrawingDetailState> {
     }
   }
 
+  Future<void> _addRecentDrawing(AddRecentDrawingEvent event, Emitter<DrawingDetailState> emit) async {
+    // User
+    User? currentUser = FirebaseAuth.instance.currentUser;
+    if (currentUser == null) {
+      return emit(DrawingDetailStateError(errorMessage: "User doesn't exist"));
+    }
+
+    String userId = currentUser.uid;
+
+    // Adding recent logic
+    FirebaseFirestore.instance.useFirestoreEmulator('192.168.1.18', 8080);
+    Query<HomeScreenData> getRecentDrawingQuery = FirebaseFirestore.instance
+        .collection('home_screens')
+        .where('user_id', isEqualTo: userId)
+        .where('project_id', isEqualTo: event.selectedProject.id)
+        .withConverter(
+          fromFirestore: HomeScreenData.fromFirestore,
+          toFirestore: HomeScreenData.toFirestore,
+        );
+
+    try {
+      QuerySnapshot<HomeScreenData> querySnapshot = await getRecentDrawingQuery.get();
+      HomeScreenData? homeScreenData = querySnapshot.docs.firstOrNull?.data();
+
+      if (homeScreenData == null) {
+        // Add data
+        final Map<String, Object?> recentData = HomeScreenData.toFirestore(HomeScreenData.fromDrawingDetail(event.drawingDetail), null);
+        await FirebaseFirestore.instance.collection('home_screens').add(recentData);
+      } else {
+        // Update existing entry
+        await FirebaseFirestore.instance
+            .collection('home_screens')
+            .doc(querySnapshot.docs.first.id)
+            .update(HomeScreenData.toFirestore(HomeScreenData.fromDrawingDetail(event.drawingDetail), null));

```

Comment:
Don't think this is correct.
This would remove all other recently viewed drawings and only record the latest one